File: packages/db/src/reconcilers/deployment-plan.ts
import type { Tx } from "../common.js";
import type { ReconcileWorkScope } from "../schema/reconcile.js";
import { enqueue } from "./enqueue.js";

const DEPLOYMENT_PLAN_KIND = "deployment-plan";

export async function enqueueDeploymentPlan(
  db: Tx,
  workspaceId: string,
  planId: string,
): Promise<ReconcileWorkScope> {
  return enqueue(db, {
    workspaceId,
    kind: DEPLOYMENT_PLAN_KIND,
    scopeType: "deployment-plan",
    scopeId: planId,
  });
}

File: apps/workspace-engine/pkg/reconcile/events/deploymentplan.go
package events

import (
	"context"

	"workspace-engine/pkg/reconcile"
)

const DeploymentPlanKind = "deployment-plan"

type DeploymentPlanParams struct {
	WorkspaceID string
	PlanID      string
}

func EnqueueDeploymentPlan(
	queue reconcile.Queue,
	ctx context.Context,
	params DeploymentPlanParams,
) error {
	return queue.Enqueue(ctx, reconcile.EnqueueParams{
		WorkspaceID: params.WorkspaceID,
		Kind:        DeploymentPlanKind,
		ScopeType:   "deployment-plan",
		ScopeID:     params.PlanID,
	})
}
